sycl/doc/extensions/experimental/sycl_ext_intel_fp_control.asciidoc

```diff
@@ -0,0 +1,176 @@
+
+* link:../experimental/sycl_ext_oneapi_properties.asciidoc[
+  sycl_ext_oneapi_properties]
+* link:../experimental/sycl_ext_oneapi_kernel_properties.asciidoc[
+  sycl_ext_oneapi_kernel_properties]
+
+== Status
+
+This is an experimental extension specification, intended to provide early access
+to features and gather community feedback. Interfaces defined in this specification
+are implemented in {dpcpp}, but they are not finalized and may change incompatibly in
+future versions of {dpcpp} without prior notice. **Shipping software products should not
+rely on APIs defined in this specification.**
+
+== Backend support status
+
+This extension is currently implemented in {dpcpp} only for Intel GPU devices and
+only when using the Level Zero backend or OpenCL backend. Also currently Intel GPU
+driver supports this extension only for Vector Backend, so feature is supported only
+for ESIMD.
+Attempting to use this extension in kernels that run on other devices or
+backends may result in undefined behavior.  Be aware that the compiler
+is not able to issue a diagnostic to warn you if this happens.
+
+== Overview
+
+Intel GPUs allow to control kernel behavior in floating point operations. This extension adds the kernel property `fp_control` which provides a way to specify the rounding mode, denorm mode and single precision floating point mode for floating point operations in a SYCL kernel.
+
+== Specification
+
+=== Feature test macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification.  An implementation supporting this extension must predefine the
+macro `SYCL_EXT_INTEL_FP_CONTROL` to one of the values defined in the table
+below.  Applications can test for the existence of this macro to determine if
+the implementation supports this feature, or applications can test the macro's
+value to determine which of the extension's features the implementation
+supports.
+
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|The APIs of this experimental extension are not versioned, so the
+ feature-test macro always has this value.
+|===
+
+=== Properties
+
+|===
+|Property|Description
+
+|`fp_control`
+|The `fp_control` kernel property provides a way to specify the rounding mode, denorm mode and single precision floating point mode for a kernel.
+|===
+
+The values that the property can be set to are a subset of the following that can be OR'ed together:
+
+[%header,cols="1,5"]
+|===
+|Mode | Description
+| rte | Round to nearest or even
+| rtp | Round towards positive infinity
+  rtn | Round towards negative infinity
+  rtz | Round towards zero
+| denorm_ftz | Denorm mode flush to zero
+| denorm_d_allow | Denorm mode double allow
+| denorm_f_allow | Denorm mode float allow
+| denorm_hf_allow | Denorm mode half allow
+| denorm_allow | Denorm mode double/float/half allow
+| float_mode_ieee | Single precision float IEEE mode
+| float_mode_alt | Single precision float ALT mode
```

Comment:
What does "alt" actually mean here?  I couldn't see an explanation here.

If it just means "not IEEE compliant", do we need to expose it at all?  Could we instead say that the default floating-point environment is not guaranteed to be IEEE-compliant, but that the presence of `float_mode_ieee` enables it?

Reply:
The meaning of the alternative mode is described here: https://github.com/intel/intel-graphics-compiler/blob/master/documentation/visa/2_datatypes.md#alt-alternative-single-floating-point-mode

Do you think it's better to provide the link to that description, or should I provide similar description in this extension?

Reply:
That description says that:
> The ALT mode is intended for legacy DirectX 9 applications

If this is only for DirectX 9 applications, should SYCL expose the concept of a "float mode" at all?

Reply:
Good point, let me ask people who requested this feature and get back to you. Thank you!

Reply:
It was confirmed that we don't need to expose these controls in sycl. Removed ieee/alt controls.